### ISSUE TITLE
Conflict with the Bob Logistic mod and its submods

### ISSUE THREAD

**Issue by keysivi** [nicolas-lang/factorio.InserterCranes#3]
Good day! I found that your mod conflicts with mods: 
Bob Logistic Krastorio2 Nuclear Locomotive https://mods.factorio.com/mod/BobLogisticKrastorio2NuclearLocomotive
Bob's Logistics mod https://mods.factorio.com/mod/boblogistics
Bob Logistic Zone Expander Radius https://mods.factorio.com/mod/BobLogisticZoneExpanderRadius

Could you please tell me - can you fix this conflict?

Thanks!

**Comment by nicolas-lang** [comment]
Hi, 
please provide:

- Logfile https://wiki.factorio.com/Log_file
- Explanation: what is the conflict - a crash or an unintended behaviour ? 
- (optional): a minimal save game to reproduce the error.

**Comment by keysivi** [comment]
[factorio-previous.log](https://github.com/nicolas-lang/factorio.InserterCranes/files/7991948/factorio-previous.log)
[factorio-current.log](https://github.com/nicolas-lang/factorio.InserterCranes/files/7991949/factorio-current.log)

**Comment by keysivi** [comment]
![Без-имени-1](https://user-images.githubusercontent.com/97015470/152285539-d21665a5-b644-4574-8960-7c0ebce8ac13.jpg)

Local list of mods that causes this error:
Inserter Cranes
Bob's Logistics mod
Bob Logistic Krastorio2 Nuclear Locomotive
Bob Logistic Zone Expander Radius
Bob's Functions Library mod
Factorio Library
Krastorio 2
Krastorio 2 Assets

**Comment by nicolas-lang** [comment]
There was an issue with recipe generation if the data model of the ingredients was non-uniform like 
```
  {
    "long-handed-inserter",
    1
  },
  {
    amount = 5,
    name = "steel-gear-wheel",
    type = "item"
  }
```
Should be fixed in 0.0.6

Thanks for the report !

**Comment by keysivi** [comment]
Thanks a lot!
